kuryr_kubernetes/cni/daemon/service.py

```diff
@@ -70,12 +72,24 @@ def _prepare_request(self):
                   params.CNI_COMMAND, params)
         return params
 
+    def _update_metrics(self, params, command, error, duration):
+        """Add a new metric value to the shared metrics dict"""
+        namespace = params.args.K8S_POD_NAMESPACE
+        name = params.args.K8S_POD_NAME
+        name = f'export-{namespace}/{name}'
+        labels = {'command': command, 'error': error}
+        with lockutils.lock(name):
+            self.metrics[name] = {'labels': labels, 'duration': duration}
+
     def add(self):
+        start_time = time.time()
```

Comment:
Given, that there is a lot of pattern repetition of start_time/time() - start time for every each piece of code, perhaps it might be a better approach to introduce a decorator for that, which simplify usage as something like:
```python
    @measure_time('ADD')
    def add(self):
```
and in inside the decorator it would simply measure whole method time, i.e.:
```python
def measure_time(command):
    def decorator(method):
        def wrapper(obj, *args, **kwargs):
            start_time = time.time()
            result = method(*args, **kwargs)
            obj._update_metrics(command, time.time() - start_time)
            return result
        return wrapper
    return decorator
```

Reply:
Thank you! This is indeed much simpler.